### ISSUE TITLE
Autoware Launching

### ISSUE THREAD

**Issue by dylan-gonzalez** [Monash-Connected-Autonomous-Vehicle/mcav-autoware#13]
1 person?

- configure all the launch files for our setups for CARLA, AWSIM, Twizy, ENV200
- configure parameters

https://github.com/tier4/aip_launcher/blob/tier4/universe/common_sensor_launch/launch/ars408.launch.xml

**Comment by AnthonyZhOon** [comment]
@Robin-Chea If you're on campus to work on this Fri I can drop in and help/discuss what we need to do

**Comment by Robin-Chea** [comment]
> @Robin-Chea If you're on campus to work on this Fri I can drop in and help/discuss what we need to do

ok ill be in from 12-5 on friday.

**Comment by Robin-Chea** [comment]
Tried to create the launch files for twizy however after using the current stuff we had we were missing some stuff eg config files that describe the links to each sensor. So instead i have created the basic struture to create these config files.
I have basically copied the sample description and launch file and configured it to the twizy sensors eg 8 camera, 1 imu, 1 lidar (VLP-16) and 1 gnss. 
I am following the guide https://autowarefoundation.github.io/autoware-documentation/pr-491/how-to-guides/integrating-autoware/creating-vehicle-and-sensor-model/

However not sure if i should include the ultrasonic sensors. 

I skipped this part for now  
https://autowarefoundation.github.io/autoware-documentation/pr-491/how-to-guides/integrating-autoware/creating-vehicle-and-sensor-model/creating-individual-params/

i skipped configuring the sensor launch for now and will do it later 
https://autowarefoundation.github.io/autoware-documentation/pr-491/how-to-guides/integrating-autoware/creating-vehicle-and-sensor-model/creating-sensor-model/#sensor-launch.

For vehicle launch desciption the config files need to be filled out.

For vehicle mesh a dae or a fbx needs to be made.

However in https://github.com/Monash-Connected-Autonomous-Vehicle/mcav_autonomy/tree/main/visualisation/vehicle_model/meshes they were able to use the stl file. will try to attempt to use this.

**Comment by Robin-Chea** [comment]
did the same but for the env200

**Comment by AbBaSaMo** [comment]
Would it be okay if I change the [repo URLs](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/blob/yen-kendrick-control/autoware.repos) in our Autoware fork to our other forks [in a new branch oc] where relevant or would there be more to it than just that? 

Is this being worked on right now as part of this issue and do I need to wait for it if I want to test out changes to sensing launch in a new branch as part of the sensing issue?

**Comment by Robin-Chea** [comment]
@AbBaSaMo Yes it should be okay to change the repo url to relevant forks, especially if it helps streamline our development process. Currently i am working on the config files for the Twizy and once its completed, ill update it in the branch for integration.

**Comment by dylan-gonzalez** [comment]
Hey @Robin-Chea can you configure the twizy and van launch files to have simulation modes too? (i.e. so we can launch the twizy and van urdf files in AWSIM for example)

So when we want to run autoware with awsim we can do something like this: `ros2 launch autoware_launch <launch_file> vehicle_model:=twizy sensor_model:=twizy_sensor_kit simulation_mode:= true map_path:=<map_path>`

I added the `simulation_mode` parameter - not sure if it already exists or not?

So ig if simulation_mode is true than it doesn't launch sd_vehicle_interface and any other hardware-specific stuff

**Comment by dylan-gonzalez** [comment]
@Robin-Chea @AbBaSaMo to finish this off this week

**Comment by Robin-Chea** [comment]
The model is now working on autoware : 

![Screenshot from 2024-01-22 12-13-16](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/assets/140467976/07be046a-557a-4c5d-8750-67ea0d0112e5)

![Screenshot from 2024-01-22 12-13-49](https://github.com/Monash-Connected-Autonomous-Vehicle/autoware/assets/140467976/09e2da3b-05dd-470f-b9b1-b6bc65b27e67)

Still need to add the ultra sensors and the gps/imu. Need to also double check the drivers for camera and velodyne
